s3pypi/core.py

```diff
@@ -89,6 +90,18 @@ def parse_distribution(path: Path) -> Distribution:
     return Distribution(name, version, path)
 
 
+def parse_distributions(paths: List[Path]) -> List[Distribution]:
+    dists = []
+    for path in paths:
+        if path.is_file():
+            dists.append(parse_distribution(path))
+        if path.is_dir():
+            for path_ in [f for f in path.iterdir() if f.is_file()]:
```

Comment:
You could also use round braces here to make this a generator rather than a list, which seems marginally better.

Reply:
I like generators. I tried to use a filter here, but, the way it was written didn't feel right, so i went with the list  comprehension. I forgot to see if I could go the generator route, good catch.